packages/dht/src/dht/routing/Router.ts

```diff
@@ -122,15 +122,20 @@ export class Router implements IRouter {
         try {
             // eslint-disable-next-line promise/catch-or-return
             logger.trace('starting to raceEvents from routingSession: ' + session.sessionId)
-            raceEvents3<RoutingSessionEvents>(session, ['routingSucceeded', 'partialSuccess', 'routingFailed', 'stopped', 'noCandidatesFound'], 10000)
-                .then(() => {
+            const eventReceived = raceEvents3<RoutingSessionEvents>(
+                session,
+                ['routingSucceeded', 'partialSuccess', 'routingFailed', 'stopped', 'noCandidatesFound'],
+                null
+            )
+            setImmediate(async () => {
+                try {
+                    await withTimeout(eventReceived, 10000)
```

Comment:
Awaiting a promise inside a setImmediate block is a dangarous patter as the promise could be rejected before the await listener is set.

Reply:
Using `executeSafePromise` wrapper to underline the fact that this promise can't reject (as it only waits events and never timeouts)